### PR TITLE
Use mochify for running code-studio tests

### PR DESCRIPTION
**Pull request by philbogle**
Change "npm test" in code-studio to use mochify.   Mochify runs the tests in an actual browser, which adds a couple seconds of compile time but allows tests to be more accurate and comprehensive (e.g. testing code that requires localStorage or sessionStorage without have to fake that functionality.)  

Caveat:  If you want to also test coverage, use "npm run coverage".  I have discovered that if the test fails, then the mochify-istanbul plugin sometimes causes it not to print the specific test failures, which makes it hard to debug

Note that I had to change include all sources from true to false when computing test coverage, and update the thresholds accordingly.  This is because mochify istanbul defaults to false and doesn't have any way of passing through a flag to change the value, based on my reading of the source code.

[x] Successful `npm test` run after changes.

```
~/code-dot-org/code-studio$ npm test
> @cdo/code-studio@0.0.1 test /Users/pbogle/code-dot-org/code-studio
> mochify test/**/*.js --transform babelify
# phantomjs:
  ..
  2 passing (5ms)

~/code-dot-org/code-studio$ npm run coverage
> @cdo/code-studio@0.0.1 coverage /Users/pbogle/code-dot-org/code-studio
> mochify test/**/*.js --transform babelify --plugin [ mochify-istanbul --report json --report lcov --dir ./coverage ] && istanbul check-coverage ./coverage/coverage-final.json
# phantomjs:
 ..
  2 passing (7ms)
```

### CODE REVIEW

**Comment by philbogle** [comment]
@islemaster PTAL
@Bjvanminnen FYI

**Comment by wjordan** [comment]
IIRC, we already use mochify and the phantomjs runner for testing `/apps` (see [runTests.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/util/runTests.js)). It looks like this is setting up mochify through a slightly different process. Would it be worth the effort to try and avoid fragmentation among our growing array of JavaScript test-runner setups by reusing shared test-setup code somewhere?

**Comment by wjordan** [comment]
(On the other hand perhaps keeping this simple and modular will be easier to maintain, since it's not much very much code as-is...)

File: code-studio/.istanbul.yml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 instrumentation:
   root: src/js
-  include-all-sources: true
+  # include-all-sources can't be true because mochify-istanbul defaults to false and doesn't respect this file.
+  include-all-sources: false
```
Comment (islemaster) [inline code-studio/.istanbul.yml:4]: Should we just remove this line from our .istanbul.yml since it's reasserting a default that we can't change?

**Comment by islemaster** [comment]
This LGTM.  I significantly prefer this to our fairly complex setup in apps.

**Comment by laurelfan** [comment]
Thanks, this totally works! (except I use jquery in my react component too...)

see:
https://github.com/code-dot-org/code-dot-org/blob/pair-programming/code-studio/test/js/pairingTest.js
(no jsdom necessary)

File: code-studio/.istanbul.yml (hunk 1 above)
Comment (philbogle) [inline code-studio/.istanbul.yml:4]: I wanted this there including the comment to discourage anyone from changing it.

Comment (philbogle) [inline code-studio/.istanbul.yml:4]: Also, I wanted to explicitly set it to a value in case the default for check coverage changes.

**Pull request merged**